### PR TITLE
Migrate to gradle

### PR DESCRIPTION
**Pull request by BlisterBoy**
I just adapted some command such as ant (because Android Studio use Gradle) and some path.

I'm currently testing that from scratch :)

### CODE REVIEW

**Comment by BlisterBoy** [comment]
This should be good !

**Comment by Ghabry** [comment]
j4 is bad, not everybody has a quad core, before was -j2

And the certificate algorithm MD5 is unsafe. You should consider SHA1

**Comment by BlisterBoy** [comment]
Corrected ^^.

**Comment by BlisterBoy** [comment]
Boost has been removed, the Readme updated.
Do you want me to delete the "rm" commands from the 1_... script?

**Comment by carstene1ns** [comment]
- boost is still used in `2_build_toolchain.sh`, this breaks the build process
- There is no real need to get java 6, as openjdk7 works here fine also

**Comment by BlisterBoy** [comment]
Okay, sorry for the boost references, I totally forgot. I also updated the Readme.

File: android/4_build_android_port.sh [hunk 1]
```diff
@@ -32,14 +32,14 @@ git clone https://github.com/Ghabry/timidity_gus.git assets/timidity
 android update project --path "." --target 1
 
 # Build
-ndk-build -j4 NDK_LIBS_OUT=./jniLibs
```
Comment (Tondorian) [inline android/4_build_android_port.sh]: maybe we can alos use -ndk-build -j$(nproc) to use all CPUs?

Comment (BlisterBoy) [inline android/4_build_android_port.sh]: That would be nice but I didn't find a portable way to get nproc :'(

Comment (Tondorian) [inline android/4_build_android_port.sh]: oh its linux specific?
i also found this:
http://stackoverflow.com/questions/2527496/how-can-i-write-a-makefile-to-auto-detect-and-parallelize-the-build-with-gnu-mak
http://stackoverflow.com/questions/2527496/how-can-i-write-a-makefile-to-auto-detect-and-parallelize-the-build-with-gnu-mak

Comment (BlisterBoy) [inline android/4_build_android_port.sh]: The script run on Linux and OSX.
Thanks for the urls, but the given solution doesn't seems to work on my OSX (and takes several minutes to terminate oO).

Comment (Tondorian) [inline android/4_build_android_port.sh]: also found: `echo $(getconf _NPROCESSORS_ONLN)`

Comment (BlisterBoy) [inline android/4_build_android_port.sh]: That seems to work on my Linux VM and my mac os, I add it :).

**Comment by BlisterBoy** [comment]
Modified the script to auto-detect the number of CPU for building, based on the Tondorian's comments.

**Comment by carstene1ns** [comment]
LGTM now. I am going to rebase this locally and we are good to go.

**Comment by Ghabry** [comment]
We found a problem with ```gradlew clean``` @BlisterBoy 
This is not the same as ```ant clean```. ant only deleted the java part of the build. gradlew also deleted the object files from the ndk-build which results in a full recompile everytime (waste of time :/)

**Comment by BlisterBoy** [comment]
Okay, I found a stackoverflow billet, I open a issue ticket.

**Pull request merged**